jobbergate-api/jobbergate_api/apps/applications/routers.py

```diff
@@ -134,22 +161,31 @@ async def application_delete(
     """
     Delete application from the database and S3 given it's id.
     """
+    logger.debug(f"Preparing to delete {application_id=} from the database and S3")
+
     where_stmt = applications_table.c.id == application_id
     get_query = applications_table.select().where(where_stmt)
+    logger.debug(f"get_query = {render_sql(get_query)}")
+
     raw_application = await database.fetch_one(get_query)
     if not raw_application:
+        message = f"Application {application_id=} not found."
+        logger.warning(message)
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND,
-            detail=f"Application {application_id=} not found.",
+            detail=message,
         )
+
     delete_query = applications_table.delete().where(where_stmt)
+    logger.debug(f"delete_query = {render_sql(delete_query)}")
+
     await database.execute(delete_query)
     try:
         del s3man_applications[application_id]
     except KeyError:
-        # We should ignore KeyErrors from the S3 manager, because the data may have already been removed
-        # outside of the API
-        pass
+        # We should ignore KeyErrors from the S3 manager, because the data may
+        # have already been removed outside of the API
+        logger.warning(f"Tried to delete {application_id=}, but it was not found on S3.")
```

Comment:
Shouldn't this be a debug as well?

Reply:
Yes

Reply:
solved in [0c61e5d](https://github.com/omnivector-solutions/jobbergate/pull/121/commits/0c61e5d42e5c10b8193ec444d45138c631f1b87a)